src/client.rs

```diff
@@ -1,96 +1,126 @@
-use crate::{
-    extract::{self, Uri},
-    types::{Excludes, RequestMethod, Status},
-};
-use anyhow::{Context, Result};
+use crate::types::{Excludes, Response, Status, Uri};
+use anyhow::{anyhow, Context, Result};
 use check_if_email_exists::{check_email, CheckEmailInput};
+use derive_builder::Builder;
 use headers::{HeaderMap, HeaderValue};
 use hubcaps::{Credentials, Github};
-use indicatif::ProgressBar;
 use regex::{Regex, RegexSet};
 use reqwest::header;
 use std::net::IpAddr;
 use std::{collections::HashSet, time::Duration};
 use tokio::time::delay_for;
 use url::Url;
 
-/// A link checker using an API token for Github links
-/// otherwise a normal HTTP client.
-pub(crate) struct Checker<'a> {
+const DEFAULT_MAX_REDIRECTS: usize = 5;
+
+pub struct Client {
     reqwest_client: reqwest::Client,
     github: Option<Github>,
     includes: Option<RegexSet>,
     excludes: Excludes,
     scheme: Option<String>,
-    method: RequestMethod,
+    method: reqwest::Method,
     accepted: Option<HashSet<reqwest::StatusCode>>,
+}
+
+/// A link checker using an API token for Github links
+/// otherwise a normal HTTP client.
+#[derive(Builder, Debug)]
+#[builder(build_fn(skip))]
+#[builder(setter(into))]
+#[builder(name = "ClientBuilder")]
+pub struct ClientBuilderInternal {
+    github_token: Option<String>,
+    includes: Option<RegexSet>,
+    excludes: Excludes,
+    #[builder(default = "5")]
```

Comment:
Can't `DEFAULT_MAX_REDIRECTS` be used here?

Reply:
I... guess so? Let me try...

Yup, you're right. 👍

Reply:
Pushed.